### PR TITLE
{Backup} Add documentation: help text details interchangeable parameter values

### PR DESCRIPTION
**Pull request by zubairabid**
**Related command**

- `az backup policy list -h`
- `az backup protectable-item list -h`
- `az backup protectable-item show -h`
- `az backup protectable-item initialize -h`
- `az backup container register -h`
- `az backup protection enable-for-azurewl`
- `az backup protection auto-enable-for-azurewl`
- `az backup protection auto-disable-for-azurewl`

**Description**

Under the option `--workload-type`, `MSSQL` and `SQLDataBase` refer to the same thing, as do `SAPHANA` and `SAPHanaDatabase`. Under `--protectable-item-type`, `SAPHanaSystem` and `HANAInstance` are duplicates as well. Removing any of the options constitutes a breaking change, and for the time being the course of action is to make it clear to the user through documentation what the various options are.

**Testing Guide**

N/A. Existing tests/commands behave the same as the only substantial changes are to documentation code.

**History Notes**

Add documentation to clarify interchangeable parameter values

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Backup

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 1]
```diff
@@ -21,18 +21,19 @@
...
 
 allowed_container_types = ['AzureIaasVM']
-allowed_workload_types = ['VM', 'AzureFileShare', 'SAPHANA', 'MSSQL', 'SAPHanaDatabase', 'SQLDataBase']
-allowed_azure_workload_types = ['MSSQL', 'SAPHANA', 'SAPASE', 'SAPHanaDatabase', 'SQLDataBase']
+allowed_workload_types = ['VM', 'AzureFileShare', 'SAPHANA', 'SAPHanaDatabase', 'MSSQL', 'SQLDataBase']
+allowed_azure_workload_types = ['MSSQL', 'SQLDataBase', 'SAPHANA', 'SAPHanaDatabase', 'SAPASE']
 allowed_backup_management_types = ['AzureIaasVM', 'AzureStorage', 'AzureWorkload']
 allowed_extended_backup_management_types = allowed_backup_management_types + ['MAB']
-allowed_protectable_item_type = ['SQLAG', 'SQLInstance', 'SQLDatabase', 'HANAInstance', 'SAPHanaDatabase', 'SAPHanaSystem']
+allowed_protectable_item_type = ['SQLAG', 'SQLInstance', 'SQLDatabase', 'HANAInstance', 'SAPHanaSystem', 'SAPHanaDatabase']
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:28]: Why do we need to exchange the order of enumeration values?

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 2]
```diff
@@ -21,18 +21,19 @@
...
 allowed_rehyd_priority_type = ['Standard', 'High']
 
 backup_management_type_help = """Specify the backup management type. Define how Azure Backup manages the backup of entities within the ARM resource. For eg: AzureWorkloads refers to workloads installed within Azure VMs, AzureStorage refers to entities within Storage account. Required only if friendly name is used as Container name."""
 container_name_help = """Name of the backup container. Accepts 'Name' or 'FriendlyName' from the output of az backup container list command. If 'FriendlyName' is passed then BackupManagementType is required."""
-workload_type_help = """Specify the type of applications within the Resource which should be discovered and protected by Azure Backup. """
+workload_type_help = """Specify the type of applications within the Resource which should be discovered and protected by Azure Backup. 'MSSQL' and 'SQLDataBase' can be used interchangeably, as can 'SAPHANA' and 'SAPHanaDatabase'."""
+protectable_item_type_help = """Specify the type of items within the Resource which should be discovered and protected by Azure Backup. 'HANAInstance' and 'SAPHanaSystem' can be used interchangeably."""
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:36]: Do we need to emphasize the recommended values in the help message? Such as: `MSSQL` is the alternative to `SQLDataBase`, and `SQLDataBase` will be deprecated in the future

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (zubairabid) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py:28]: There is no need, it just happened while reverting an older commit. Should I change it back to what it was earlier, or is this fine?

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 2 above)
Comment (zubairabid) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py:36]: I don't want to add the deprecation message yet, as we'll need to discuss it with the team first. So once that's finalized we could emphasize recommended values, until then interchangeable should be fine.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py:36]: OK, got it

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py:28]: Please roll back to reduce the unnecessary modifications

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 3]
```diff
@@ -21,18 +21,19 @@
 # ARGUMENT DEFINITIONS
 
 allowed_container_types = ['AzureIaasVM']
-allowed_workload_types = ['VM', 'AzureFileShare', 'SAPHANA', 'MSSQL', 'SAPHanaDatabase', 'SQLDataBase']
-allowed_azure_workload_types = ['MSSQL', 'SAPHANA', 'SAPASE', 'SAPHanaDatabase', 'SQLDataBase']
+allowed_workload_types = ['VM', 'AzureFileShare', 'SAPHANA', 'SAPHanaDatabase', 'MSSQL', 'SQLDataBase']
+allowed_azure_workload_types = ['MSSQL', 'SQLDataBase', 'SAPHANA', 'SAPHanaDatabase', 'SAPASE']
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:25]: Same as https://github.com/Azure/azure-cli/pull/24535/files#r1012573852

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (zubairabid) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: Done.

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**